File: sesion8/web3/controllers/task_controller.py
from web3.models.task import Task

from flask import redirect

import json

class TaskController:

    tasks = []

    def list():
        pass

    def get(id):
        pass

    def create(req):
        print("Creando tarea")
        form_data = req.form
        task_name = form_data['taskname']

        id = len(TaskController.tasks)
        new_task = Task(id, task_name)

        TaskController.tasks.append(new_task)
        return redirect('/')


    def update(id, req):

        body = json.loads(req.data.decode())

        index = 0
        for idx, task in enumerate(TaskController.tasks):
            if int(task.id) == int(id):
                index = idx

        if "title" in body: 
            title = body["title"]
            TaskController.tasks[index].title = title

        if "status" in body: 
            status = body["status"]
            TaskController.tasks[index].status = status

        return "OK"

    def delete(id):
        for task in TaskController.tasks:
            if int(task.id) == int(id):
                print(f"Elimino: {task}")
                TaskController.tasks.remove(task)
        return redirect('/')

File: sesion35/backend/controllers/task_controller.js
 
const { TaskModel } = require('../db')

class TaskController {

    static create(req, res) {

        let data = req.body

        TaskModel.create(data)
            .then(data => {
                res.send(data)
            })
            .catch(err => {
                res.status(400).send({
                    message: err.message
                })
            })

    }

    static findAll(req, res) {
        var query = TaskModel.find({})

        query.exec(function (err, tasks) {
            if (err) {
                res.sendStatus(404)
            } else {
                res.send(tasks)
            }
        })
    }

    static update(req, res) {
        let pk = req.params.pk
        let payload = req.body
        let query = TaskModel.findByIdAndUpdate(pk, payload)

        query.exec(function(err, task) {
            if (err) {
                res.sendStatus(404)
            } else {
                res.send(task)
            }
        })

    }

    static delete(req, res) {
        let pk = req.params.pk
        let query = TaskModel.findByIdAndDelete(pk)
        query.exec(function(err, task) {
            if (err) {
                res.sendStatus(404)
            } else {
                res.send(task)
            }
        })
    }

}

module.exports = { TaskController }